File: C1.UWP.FlexChart/VB/LineMarker/ViewModel/LineMarkerViewModel.vb
Imports C1.Xaml.Chart.Interaction
Imports System.Collections.Generic
Imports System.Linq
Imports System

Public Class LineMarkerViewModel
    Const Count As Integer = 300
    Private rnd As New Random()

    Public ReadOnly Property Items() As List(Of DataItem)
        Get
            Dim data As New List(Of DataItem)()
            Dim [date] As New DateTime(2016, 1, 1)
            Dim i As Integer = 0
            While i < Count
                Dim item As New DataItem() With {
                    .[Date] = [date].AddDays(i),
                    .Input = rnd.[Next](10, 20),
                    .Output = rnd.[Next](0, 10)
                }
                data.Add(item)
                i += 1
            End While

            Return data
        End Get
    End Property

    Public ReadOnly Property LineType() As List(Of String)
        Get
            Return [Enum].GetNames(GetType(LineMarkerLines)).ToList()
        End Get
    End Property

    Public ReadOnly Property LineMarkerInteraction() As List(Of String)
        Get
            Return [Enum].GetNames(GetType(LineMarkerInteraction)).ToList()
        End Get
    End Property

    Public ReadOnly Property LineMarkerAlignments() As Dictionary(Of String, LineMarkerAlignment)
        Get
            Dim alignments As New Dictionary(Of String, LineMarkerAlignment)()
            alignments.Add(Strings.Auto, LineMarkerAlignment.Auto)
            alignments.Add(Strings.Right, LineMarkerAlignment.Right)
            alignments.Add(Strings.Left, LineMarkerAlignment.Left)
            alignments.Add(Strings.Bottom, LineMarkerAlignment.Bottom)
            alignments.Add(Strings.Top, LineMarkerAlignment.Top)
            alignments.Add(Strings.LeftBottom, LineMarkerAlignment.Left Or LineMarkerAlignment.Bottom)
            alignments.Add(Strings.LeftTop, LineMarkerAlignment.Left Or LineMarkerAlignment.Top)
            Return alignments
        End Get
    End Property

    Public ReadOnly Property CanDrag() As List(Of String)
        Get
            Return New List(Of String)() From {
                Strings.[True],
                Strings.[False]
            }
        End Get
    End Property
End Class

File: C1.UWP.FlexChart/CS/LineMarker/ViewModel/LineMarkerViewModel.cs
﻿using System;
using System.Linq;
using System.Collections.Generic;
using C1.Xaml.Chart.Interaction;

namespace LineMarkerSample
{
    public class LineMarkerViewModel
    {
        const int Count = 300;
        Random rnd = new Random();

        public List<DataItem> Items
        {
            get
            {
                List<DataItem> items = new List<DataItem>();
                DateTime date = new DateTime(2016, 1, 1);
                for (var i = 0; i < Count; i++)
                {
                    var item = new DataItem()
                    {
                        Date = date.AddDays(i),
                        Input = rnd.Next(10, 20),
                        Output = rnd.Next(0, 10)
                    };
                    items.Add(item);
                }

                return items;
            }
        }

        public List<string> LineType
        {
            get
            {
                return Enum.GetNames(typeof(LineMarkerLines)).ToList();
            }
        }

        public List<string> LineMarkerInteraction
        {
            get
            {
                return Enum.GetNames(typeof(LineMarkerInteraction)).ToList();
            }
        }

        public Dictionary<string, LineMarkerAlignment> LineMarkerAlignments
        {
            get
            {
                var alignments = new Dictionary<string, LineMarkerAlignment>();
                alignments.Add(Strings.Auto, LineMarkerAlignment.Auto);
                alignments.Add(Strings.Right, LineMarkerAlignment.Right);
                alignments.Add(Strings.Left, LineMarkerAlignment.Left);
                alignments.Add(Strings.Bottom, LineMarkerAlignment.Bottom);
                alignments.Add(Strings.Top, LineMarkerAlignment.Top);
                alignments.Add(Strings.LeftBottom, LineMarkerAlignment.Left | LineMarkerAlignment.Bottom);
                alignments.Add(Strings.LeftTop, LineMarkerAlignment.Left | LineMarkerAlignment.Top);
                return alignments;
            }
        }

        public List<string> CanDrag
        {
            get
            {
                return new List<string>() { Strings.True, Strings.False };
            }
        }
    }
}
